src/core/source/ireplay.ts

```diff
@@ -0,0 +1,316 @@
+          this._updateId.bind(this))
+      }
+      this._checkPromise
+      .then(buffer => {
+        resolve(Number(buffer));
+      }).catch(err => reject(err));
+    })
+  }
+
+  setReplaytime(buffer: number): Promise<SourceReplay> {
+    return new Promise((resolve, reject) => {
+      if (typeof buffer === 'number') {
+        if(this._isItemCall){
+          Logger.warn('sourceWarning', 'setReplaytime', true)
+          iItem.set('prop:presproperty:buffer', String(buffer), this._id)
+            .then(val => {
+              resolve(this);
+            });
+        } else {
+          iItem.wrapSet('prop:presproperty:buffer', String(buffer),
+            this._srcId, this._id, this._updateId.bind(this))
+            .then(val => {
+              resolve(this);
+            });
+        }
+      } else {
+        reject(Error('Invalid parameter. setReplaytime method only accepts buffer as a number.'));
+      }
+    })
+  }
+
+  startReplay(): Promise<SourceReplay> {
+    return new Promise((resolve, reject) => {
+      if(this._isItemCall){
+        Logger.warn('sourceWarning', 'startReplay', true)
+        iItem.set('prop:ReplayActive', '1', this._id)
+          .then(val => {
+            resolve(this);
+          });
+      } else {
+        iItem.wrapSet('prop:ReplayActive', '1',
+          this._srcId, this._id, this._updateId.bind(this))
+          .then(val => {
+            resolve(this);
+          });
+      }
+    })
+  }
+
+  stopReplay(): Promise<SourceReplay> {
+    return new Promise((resolve, reject) => {
+      if(this._isItemCall){
+        Logger.warn('sourceWarning', 'stopReplay', true)
+        iItem.set('prop:ReplayActive', '0', this._id)
+          .then(val => {
+            resolve(this);
+          });
+      } else {
+        iItem.wrapSet('prop:ReplayActive', '0',
+          this._srcId, this._id, this._updateId.bind(this))
+          .then(val => {
+            resolve(this);
+          });
+      }
+    })
+  }
+
+  getReplayState(): Promise<number> {
+    return new Promise((resolve, reject) => {
+      if(this._isItemCall){
+        Logger.warn('sourceWarning', 'getReplayState', true)
+        this._checkPromise = iItem.get('prop:ReplayActive', this._id)
+      } else {
+        this._checkPromise = iItem.wrapGet('prop:ReplayActive', this._srcId, this._id,
+          this._updateId.bind(this))
+      }
+      this._checkPromise
+      .then(activeState => resolve(Number(activeState)))
+      .catch(err => reject(err));
+    })
+  }
+
+
+  isAutostartOnSceneLoad(): Promise<boolean> {
+    return new Promise(resolve => {
+      if(this._isItemCall){
+        Logger.warn('sourceWarning', 'isAutostartOnSceneLoad',  true)
+        this._checkPromise = iItem.get('prop:StartOnLoad', this._id)
+      } else {
+        this._checkPromise = iItem.wrapGet('prop:StartOnLoad', this._srcId,
+          this._id, this._updateId.bind(this).bind(this))
```

Comment:
double `bind(this)` seems to be a copy-paste mistake

Reply:
Nope, upon checking, I've copied correctly, alright. It's the original that's the culprit. :D Committed the necessary changes.